### DISCUSSION TITLE
[react-native] Property 'nativeEvent' does not exist on type 'LayoutChangeEvent'

### DISCUSSION THREAD

**Discussion by luiscurini** [DefinitelyTyped/DefinitelyTyped#54680] [Issues with a @types package]
- [x]  I tried using the @types/xxxx package and had problems.
- [x]  I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x]  I understand that @types package versions do not follow SemVer.
- [x]  I have a question that is inappropriate for StackOverflow. (Please ask any appropriate questions there).
- [x] Mention the authors (see Definitions by: in index.d.ts) so they can respond.
 
@types/react-native@0.63.52

When I try to use  get the layout like

```
handleHeaderLayout = (event: LayoutChangeEvent) => {
    this.headerHeight = event.nativeEvent.layout.height
  }
``` 

tsc complain about this: 
` Property 'nativeEvent' does not exist on type 'LayoutChangeEvent'.`

Am I missing something?

@AntoineDoubovetzky @alloy @Esemesek @zusinShinpei @jeremybarbet

**Comment by nnnnoel** [comment]
use NativeSyntheticEvent<LayoutChangeEvent>

**Reply by luiscurini** [reply]
I'm using it on this on 

```
 <View style={styles.container} onLayout={this.handleHeaderLayout}>children</View>
```

The definition in ViewProps is: 
```
 /**
     * Invoked on mount and layout changes with
     *
     * {nativeEvent: { layout: {x, y, width, height}}}.
     */
    onLayout?: (event: LayoutChangeEvent) => void;
```

and LayoutChangeEvent is: 

```
// @see TextProps.onLayout
export type LayoutChangeEvent = NativeSyntheticEvent<{ layout: LayoutRectangle }>;
```

Which is already a NativeSyntheticEvent. Also in the tests, it's called event.nativeEvent.

```
class TextTest extends React.Component {
    handleOnLayout = (e: LayoutChangeEvent) => {
        testNativeSyntheticEvent(e);

        const x = e.nativeEvent.layout.x; // $ExpectType number
        const y = e.nativeEvent.layout.y; // $ExpectType number
        const width = e.nativeEvent.layout.width; // $ExpectType number
        const height = e.nativeEvent.layout.height; // $ExpectType number
    };
}
```

I'm still confused why nativeEvent doesn't exist. Any other hints @nnnnoel
